docs/source/UserGuide/GettingStarted/GettingUserIdentity/index.rst

```diff
@@ -61,3 +61,20 @@ If another non-default user group is needed, the command becomes::
   $ dirac-proxy-init -g <user_group>
 
 where ``user_group`` is the desired DIRAC group name for which the user is entitled.
+
+Token authorization
+-------------------
+
+Starting with the v7r4 version in DIRAC, it is possible to authorize users through third party Identity Providers (IdP),
+such as EGI Checkin [https://www.egi.eu/services/check-in/] or WLCG IAM (https://indico.cern.ch/event/739896/contributions/3497694/attachments/1905332/3146590/IAM-WLCG-AuthZ-Fermilab-10092019.pdf).
+To do this, you do not need to have a certificate if you use a terminal, the main thing is that you must be registered in one of the supported IdP. The registration process is different for each IdP.
+
+Once your account is created, you will be able to register with DIRAC using the `dirac-login` command that will return tokens that will be used to access the services::
+
+  dirac-login -g <user_group>
+
+But since not all services currently support tokens, you can get a proxy if you use the "--proxy" key::
+
+  dirac-login -g <user_group> --proxy
```

Comment:
the result is the same - a proxy with a group.

I can add a "--use-certificate" key to generate a proxy completely on the client side (as dirac-proxy-init does).

dirac-proxy-init analog:

```
dirac-login --use-certificate
```

Login without local certificat to get proxy:

```
dirac-login --proxy
```

To get token:

```
dirac-login
```

What do you think?

Reply:
Yes, I think it would make sense -- one command to rule them all.

Reply:
Can this command be made smarter so users don't need to understand the weird transitional mess?

* Perhaps `--proxy` should be the default for now and `--no-proxy` can be used instead if you really don't want one.
* `--use-certificate` is enabled by default if one is available (`--use-certificate`/`--no-use-certificate` can be used to make it explicit)

Reply:
In order to avoid clutter when starting to use tokens, I left `dirac-proxy-init` (which everyone knows well) and added a new `dirac-login` method, which is fundamentally differs in the method of authorization (via authorization server).

Some notes:

- At authorization we can receive tokens or proxies.
- I think we can assume that very soon all services will be tornadoed and DIRAC will still have something to use.
- OAuth2 framework uses tokens, if you remember our discussions and agree to do everything as standard as possible, then I think the tokens should be the default for dirac-login.
- on the client side there may be requests to non-DIRAC resources with a proxy or token (received through DIRAC AS).

IMHO:
> Perhaps --proxy should be the default for now and --no-proxy can be used instead if you really don't want one.

I should be --proxy and --tokens.

I'll think about what the script can decide without asking the user, so as not to be too clever.

Reply:
I did not find the perfect solution, but given the wishes did the following:

Choose authorization way: `--use-certificate`/`--use-diracas`
Choose response type: `--proxy`/`--token`
By default: `--use-certificate` `--proxy`

1. create proxy locally with certificate:
```
dirac-login
# will return proxy
```

2. Login through DIRAC Authorization Server:
```
dirac-login --use-diracas
# will return proxy
``` 
```
dirac-login --token
# will return token
```